politeiawww/cmd/politeiawwwcli/commands/newdcc.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2017-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package commands
+
+import (
+	"bufio"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/decred/politeia/politeiad/api/v1/mime"
+	cms "github.com/decred/politeia/politeiawww/api/cms/v1"
+	www "github.com/decred/politeia/politeiawww/api/www/v1"
+	"github.com/decred/politeia/util"
+)
+
+// NewDCCCmd submits a new dcc.
+type NewDCCCmd struct {
+	Args struct {
+		Type        uint     `positional-arg-name:"type"`            // 1 for Issuance, 2 for Revocation
+		Attachments []string `positional-arg-name:"attachmentfiles"` // DCC attachment files
+	} `positional-args:"true" optional:"true"`
+	Type          string ``
+	NomineeUserID string `long:"nomineeuserid" optional:"true" description:"The UserID of the Nominated User"`
+	Statement     string `long:"statement" optional:"true" description:"Statement in support of the DCC"`
+	Domain        string `long:"domain" optional:"true" description:"The domain of the nominated user"`
+}
+
+// Execute executes the new dcc command.
+func (cmd *NewDCCCmd) Execute(args []string) error {
+
+	// Check for a valid DCC type
+	if int(cmd.Args.Type) <= 0 || int(cmd.Args.Type) > 2 {
+		return errInvalidDCCType
+	}
+
+	// Check for user identity
+	if cfg.Identity == nil {
+		return errUserIdentityNotFound
+	}
+
+	// Get server public key
+	vr, err := client.Version()
+	if err != nil {
+		return err
+	}
+	var domainType int
+	if cmd.Statement == "" || cmd.NomineeUserID == "" || cmd.Domain == "" {
+		reader := bufio.NewReader(os.Stdin)
+		if cmd.Statement == "" {
+			fmt.Print("Enter your statement to support the DCC: ")
+			cmd.Statement, _ = reader.ReadString('\n')
+		}
+		if cmd.NomineeUserID == "" {
+			fmt.Print("Enter the nominee user id: ")
+			cmd.NomineeUserID, _ = reader.ReadString('\n')
+		}
+		if cmd.Domain == "" {
+			for {
```

Comment:
My spiddy senses went off when I saw an endless loop. This looks correct but it could stand to check `update` in the declaration.